cmd/wait-for-files-to-exist/main.go

```diff
@@ -35,85 +36,120 @@ func main() {
 	}
 	flag.Parse()
 
-	if flag.NArg() < 1 {
-		exit("must specify a file to watch")
-	}
-	if flag.NArg() > 1 {
-		exit("only one file can be specified")
-	}
-	fileToWaitFor := flag.Arg(0)
-
+	// Set up a timeout context if requested.
 	if *timeout > 0 {
 		timeoutCtx, cancel := context.WithTimeout(ctx, *timeout)
 		defer cancel()
 
 		ctx = timeoutCtx
 	}
 
-	w, err := fsnotify.NewWatcher()
-	if err != nil {
-		exit("creating a new watcher: %s", err)
+	if flag.NArg() < 1 {
+		exit("must specify at-least one file to watch")
 	}
-	defer w.Close()
+	fileToWaitFor := flag.Args()
 
-	// Start listening for events.
-	errChan := make(chan error)
-	go fileLoop(ctx, w, fileToWaitFor, errChan)
+	// Validate and exit early if any file is invalid.
+	for _, f := range fileToWaitFor {
+		if err := validateFileToWaitFor(f); err != nil {
+			exit("%w", err)
+		}
+	}
 
+	// Create one watcher per file.
+	var wg sync.WaitGroup
+	wg.Add(len(fileToWaitFor))
+	type fileResult struct {
+		file string
+		err  error
+	}
+	resultCh := make(chan fileResult, len(fileToWaitFor))
+	for _, f := range fileToWaitFor {
+		printOutput("waiting for file %q to exist", f)
+		go func(path string) {
+			defer wg.Done()
+			resultCh <- fileResult{file: path, err: waitForFile(ctx, path)}
+		}(f)
+	}
+
+	go func() {
+		wg.Wait()
+		close(resultCh)
+	}()
+
+	for res := range resultCh {
+		if res.err != nil {
+			exit("error waiting for file %q: %w", res.file, res.err)
+		}
+		printOutput("file %q exists!", res.file)
+	}
+}
+
+func validateFileToWaitFor(fileToWaitFor string) error {
 	st, err := os.Lstat(fileToWaitFor)
 
 	switch {
 	case os.IsNotExist(err):
 		// File does not exist, so we need to watch the parent directory for the file creation.
 		// Ensure to watch the real path rather than a symlinked path which does not work properly with
 		// inotify.
-		dirPath, err := filepath.EvalSymlinks(filepath.Dir(fileToWaitFor))
+		_, err = filepath.EvalSymlinks(filepath.Dir(fileToWaitFor))
 		if err != nil {
-			exit(
-				"failed to evaluate any symlinks to read real directory for %q: %v",
+			return fmt.Errorf(
+				"failed to evaluate any symlinks to read real directory for %q: %w",
 				fileToWaitFor,
 				err,
 			)
 		}
```

Comment:
Let's add a return in here for better readability.
```suggestion
		}
		
		return nil
```